app/views/pages/use-cases/using-websockets.liquid

```diff
@@ -0,0 +1,211 @@
+  function conversation = 'modules/chat/lib/queries/conversations/find_by_participant', id: room_id, participant_id: current_profile.id
+
+  if conversation
+    echo 'true'
+  else
+    echo 'false'
+  endif
+%}
+{% endraw %}
+```
+In the pOS Marketplace Template, each user has a profile and the profile is linked with a conversation. If the conversation is found then we return `true` to confirm that the user can subscribe to this room.
+
+### Step 6: Send messages
+
+To send a message, you need to call the `send` method on the subscription object you have created. This method takes Object as an argument.
+
+```javascript
+// modules/chat/src/js/chat.js
+
+module.sendMessage = (message) => {
+  let messageData = {
+    message: encodeHtml(message),
+    author_id: module.settings.currentUserId,
+    sender_name: module.settings.messageInput.getAttribute('data-from-name'),
+    created_at: new Date(),
+    create: true
+  };
+
+  module.channel.send(messageData);
+};
+```
+
+### Step 7: Set server callback on received message
+
+You may want to trigger some code when the server receives a message from a client. To do so, you need to create a partial that will be called when the message is received. This file should be stored in `views/partials/channels/:channel_name/receive.liquid`. In this example, we use code from the [pOS Marketplace Template](/developer-guide/pos-marketplace-template/core-concepts) and the channel is called `conversate`.
+
+`context.params` contains all fields that have been sent in the previous step.
+
+```liquid
+{% raw %}
+{%- comment -%}
+modules/chat/public/views/partials/channels/conversate/receive.liquid
+{%- endcomment -%}
+
+{% liquid
+  function current_profile = 'lib/current_organization_profile', user_id: context.current_user.id
+  assign room_id = context.params.room_id
+  function conversation = 'modules/chat/lib/queries/conversations/find_by_participant', id: room_id, participant_id: current_profile.id
+  assign message_safe = context.params.message | raw_escape_string
+  assign object = '{}' | parse_json
+  hash_assign object['conversation_id'] = conversation.id
+  hash_assign object['autor_id'] = current_profile.id
+  hash_assign object['message'] = message_safe
+  function message = 'modules/chat/lib/commands/messages/create', object: object
+  if message.valid != true
+    log message, 'ERROR receive message'
+  else
+    function res = 'modules/chat/lib/commands/conversations/mark_unread', conversation: conversation, current_profile: current_profile
+  endif
+%}
+{% endraw %}
+```
+In this example, we are retrieving the conversation object and creating a message object to persist data in the database.
+
+### Other examples
+
+In the [pOS Marketplace Template](/developer-guide/pos-marketplace-template/core-concepts), we have also implemented a notifications functionality that uses WebSockets to inform user about unread messages. You can find it in the file `modules/chat/src/js/notifications.js`.
```

Comment:
This could be a link to the file in github repo `modules/chat/src/js/notifications.js`

Reply:
Thank you! I added the link.